### PR TITLE
V4 unit test overhaul

### PR DESCRIPTION
**Pull request by ThomasGalesloot**
# Pull Request Key Information

## Reason for change

Many unit testing were failing due to being outdated

## What was changed

All failing unit tests were updated to be passing on the current codebase. Unit tests will be kept up to date moving forward.

## Intended Purpose

Establish a stable baseline of passing unit tests.

## Level of version change required

N/A

## Testing Completed

N/A

## Target Issue

N/A

## Additional Information

N/A

### CODE REVIEW

File: LDAR_Sim/src/sensors/default_sensor.py [hunk 1]
```diff
@@ -18,24 +18,32 @@
...
 
 class DefaultSensor:
-    def __init__(self, mdl: float, quantification_error: float) -> None:
-        self._mdl: float = mdl
+    def __init__(self, mdl: Union[list[float], float], quantification_error: float) -> None:
+        if isinstance(mdl, (float, int)):
+            self._mdl: float = mdl
+        elif isinstance(mdl, list):
+            self._mdl: float = mdl[0]
+            if len(mdl) == 4:
```
Comment (SEAJang) [inline LDAR_Sim/src/sensors/default_sensor.py:32]: Why are we checking that the default sensor MDL length is equal to 4?
The default sensor is a threshold. It makes no sense to do this check for the default sensor. 
If anything if the default sensor is given a list of floats with length greater than 1, printing out an error might be better.

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_sources/test_generate_emissions.py [hunk 2]
```diff
@@ -74,39 +92,56 @@ def test_001_same_start_end_date_behavior(
...
+            repair_delay_dataframe={},
+        )
     emis_counts: list[int] = []
     for sim, emissions in src._generated_emissions.items():
         emis_counts.append(len(emissions))
     assert average(emis_counts) == pytest.approx(expected_res, rel=0.1)
 
 
-# TODO : fix up below test
 # def test_001_validate_randomness(self):
```
Comment (SEAJang) [inline LDAR_Sim/testing/unit_testing/test_virtual_world/test_sources/test_generate_emissions.py:110]: This was my bad. But we may want to fix up this test.

File: LDAR_Sim/testing/unit_testing/test_sensors/test_site_level_sensors/test_detect_emissions.py [hunk 3]
```diff
@@ -13,19 +13,13 @@
...
 ) -> None:
     mock_site: Site = Site()
-    mdl: float = (
-        sensor_info_for_default_equipment_group_level_sensor_construction_testing["mdl"]
-    )
-    qe: float = (
-        sensor_info_for_default_equipment_group_level_sensor_construction_testing["QE"]
-    )
-    sensor = DefaultSiteLevelSensor(mdl, qe)
+    mdl: float = sensor_info_for_default_equipment_group_level_sensor_construction_testing["mdl"]
```
Comment (SEAJang) [inline LDAR_Sim/testing/unit_testing/test_sensors/test_site_level_sensors/test_detect_emissions.py:20]: could update to use constants here.

**Review by SEAJang — APPROVED** [review]
A few locations could use constants, but that'll be touched on in the constants overhaul soon, so don't need to make those changes.
A quick comment in the default sensor, otherwise I don't see any issues with the updated unit tests. 

Great update. 

I am going to suggest for the PR merge to do a quick interactive rebase on the git history such that the sensor changes are separated out. Everything else can go into a single commit - so have 2-3 commits for this unit test overhaul instead of the single squash and merge.

File: LDAR_Sim/testing/unit_testing/test_virtual_world/test_sources/test_generate_emissions.py (hunk 2 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/testing/unit_testing/test_virtual_world/test_sources/test_generate_emissions.py:110]: Can take a look

File: LDAR_Sim/testing/unit_testing/test_sensors/test_site_level_sensors/test_detect_emissions.py (hunk 3 above)
Comment (ThomasGalesloot) [reply LDAR_Sim/testing/unit_testing/test_sensors/test_site_level_sensors/test_detect_emissions.py:20]: lets do this with the constants update

**Pull request merged**